### ISSUE TITLE
python27 is dropped without guards in setup.py

### ISSUE THREAD

**Issue by jonwright** [silx-kit/silx#3116]
Assuming you are dropping 2.7, then please

- See https://packaging.python.org/guides/dropping-older-python-versions/

- Your setup.py misses python_requires and incorrectly lists python2.7 in classifiers.

The pre-release should bug out with pip and leave the user with the last good silx for 2.7.

```
Successfully built silx
Installing collected packages: silx
  Found existing installation: silx 0.11.0
    Uninstalling silx-0.11.0:
      Successfully uninstalled silx-0.11.0
Successfully installed silx-0.13.0b2

(py27_64) C:\Users\wright>python
Python 2.7.15 (default, Jul  2 2019, 00:55:11) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import silx.test
>>> dir(silx.test)
['__authors__', '__builtins__', '__date__', '__doc__', '__file__', '__license__', '__name__', '__package__', '__path__', 'logger', 'logging', 'run_tests', 'suite', 'test_options', 'unittest', 'utils']
>>> silx.test.run_tests()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\miniconda3\envs\py27_64\lib\site-packages\silx\test\__init__.py", line 96, in run_tests
    if not runner.run(suite()).wasSuccessful():
  File "c:\miniconda3\envs\py27_64\lib\site-packages\silx\test\__init__.py", line 63, in suite
    from .test_sx import suite as test_sx_suite
  File "c:\miniconda3\envs\py27_64\lib\site-packages\silx\test\test_sx.py", line 40, in <module>
    from silx.gui.colors import rgba
  File "c:\miniconda3\envs\py27_64\lib\site-packages\silx\gui\colors.py", line 715
    def setGammaNormalizationParameter(self, gamma: float) -> None:
                                                  ^
SyntaxError: invalid syntax
```

**Comment by t20100** [comment]
Indeed

**Comment by t20100** [comment]
Closed by PR #3119 on branch 0.13
